general_itests/fake_simple_service/Makefile

```diff
@@ -15,7 +15,15 @@
 DOCKER_TAG ?= fake_simple_service-$(USER)-dev
 export DOCKER_TAG
 
+ifeq ($(findstring .yelpcorp.com,$(shell hostname -f)), .yelpcorp.com)
+	docker_registry = docker-dev.yelpcorp.com/
+	pip_index_url = https://pypi.yelpcorp.com/simple
+else
+	docker_registry = ""
+	pip_index_url = https://pypi.python.org/simple
+endif
+
 .PHONY: cook-image
 
 cook-image:
-	docker build -t "$$DOCKER_TAG" .
+	docker build --build-arg PIP_INDEX_URL=$(pip_index_url) --build-arg DOCKER_REGISTRY=$(docker_regsitry) -t "$$DOCKER_TAG" .
```

Comment:
this was the tricky bit -- something was wrong with the interpolation for the docker_registry.  it was using bash env variables, and I switched it to using variables in the Makefile itself.